### PR TITLE
update ironic-image required tests to Prow tests

### PR DESCRIPTION
**Pull request by tuminoid**
Update ironic-image required tests to Prow tests. Remove optional tests as they conflict with required tests, and make little sense to be available cross-releases.

### CODE REVIEW

**Comment by tuminoid** [comment]
/hold

Optional tests needed to be removed as they conflict with required tests. IMO they also make little sense anyways. Thoughts?

/cc @mboukhalfa @kashifest @lentzi90

File: prow/manifests/overlays/metal3/config.yaml [hunk 1]
```diff
@@ -2921,39 +2921,6 @@ presubmits:
                 value: "TRUE"
             image: docker.io/pipelinecomponents/markdownlint:0.13.0@sha256:9c0cdfb64fd3f1d3bdc5181629b39c2e43b6a52fc9fdc146611e1860845bbae0
             imagePullPolicy: Always
-    # name: {job_prefix}-{image_os}-e2e-integration-test-{capm3_target_branch}
-    - name: metal3-centos-e2e-integration-test-main
-      agent: jenkins
-      always_run: false
-      optional: true
```
Comment (lentzi90) [inline prow/manifests/overlays/metal3/config.yaml:2928]: These are the actual prow job definitions. Without them you won't be able to trigger anything through prow.
The required status checks above are just things tide should make sure are there before merging. I.e. prow knows nothing about them except that they must be there

**Review by lentzi90 — CHANGES_REQUESTED** [review]
The prow jobs have to stay if we want to trigger the tests through prow.

File: prow/manifests/overlays/metal3/config.yaml (hunk 1 above)
Comment (tuminoid) [reply prow/manifests/overlays/metal3/config.yaml:2928]: Oh yes, of course. However they cannot be optional, and need to be branch by branch defined then, as they cannot be listed like this and be required. Prow won't allow that.

**Comment by tuminoid** [comment]
OK, fixed the confusion with the jenkins contexts and prow contexts. How about now?

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by Rozzii — APPROVED** [review]
/lgtm

**Pull request merged**